torch/quantization/fx/quantize.py

```diff
@@ -0,0 +1,765 @@
+                weight = load_arg(quantized=True)(self.linear_node.args[1])
+                bias = None
+                other_args = load_arg(quantized=False)(self.linear_node.args[1:])
+                if len(self.linear_node.args) > 2:
+                    bias = load_arg(quantized=False)(self.linear_node.args[2])
+                    other_args = other_args[1:] # remove the bias argument
+                else:
+                    assert 'bias' in kwargs, \
+                        'expect bias provided as a keyword argument when it is not a positional argument'
+                    bias = kwargs['bias']
+                    kwargs.pop('bias')
+                prepack_args = [weight, bias]
+                packed_weight = quantizer.quantized_graph.call_function(
+                    torch.ops.quantized.linear_prepack, prepack_args, {})
+                # construct linear input
+                linear_input = load_arg(quantized=True)(self.linear_node.args[0])
+                activation_post_process = \
+                    quantizer.activation_post_process_map[self.linear_node.name]
+                scale, zero_point, _ = get_qparams(activation_post_process)
+                qlinear_args = [linear_input, packed_weight, scale, zero_point]
+                return quantizer.quantized_graph.call_function(
+                    torch.ops.quantized.linear, qlinear_args, kwargs)
+
+# these ops have quantized equivalents that do not need any extra information
+@register_quant_pattern(torch.nn.AdaptiveAvgPool2d)
+@register_quant_pattern(torch.nn.AvgPool2d)
+@register_quant_pattern(torch.nn.Dropout)
+@register_quant_pattern(torch.nn.MaxPool2d)
+@register_quant_pattern(torch.nn.ReLU)
+@register_quant_pattern(torch.nn.ReLU6)
+@register_quant_pattern(torch.nn.functional.adaptive_avg_pool2d)
+@register_quant_pattern(torch.nn.functional.dropout)
+@register_quant_pattern(torch.nn.functional.max_pool2d)
+@register_quant_pattern(torch._C._nn.avg_pool2d)
+@register_quant_pattern(torch.flatten)
+@register_quant_pattern(torch.transpose)
+@register_quant_pattern(torch.mean)
+@register_quant_pattern(torch.unsqueeze)
+@register_quant_pattern(operator.getitem)
+@register_quant_pattern(operator.floordiv)
+@register_quant_pattern('chunk')
+@register_quant_pattern('contiguous')
+@register_quant_pattern('mean')
+@register_quant_pattern('reshape')
+@register_quant_pattern('shape')
+@register_quant_pattern('size')
+@register_quant_pattern('view')
+class CopyNode(QuantizerPattern):
+    def convert(self, quantizer, node, load_arg, debug=False):
+        return quantizer.quantized_graph.node_copy(node, load_arg(quantized=None))
+
+class DefaultQuant(QuantizerPattern):
+    def convert(self, quantizer, node):
+        assert self.all_tensors
+        return quantize(quantizer, node)
+
+# 2. Post Training Dynamic Quantizatoin Patterns
+@register_dynamic_pattern(torch.nn.Linear)
+@register_dynamic_pattern(torch.nn.functional.linear)
+class DynamicLinear(QuantizerPattern):
```

Comment:
For dynamic quant, can we run subset of the graph to run only weight observers?

Reply:
This is not handled yet, but should still be doable.